Gordon360/Models/ViewModels/BasicInfoViewModel.cs

```diff
@@ -74,10 +76,23 @@ public string GetFirstNameFromUsername()
 
         public string GetLastNameFromUsername()
         {
-            return UserName.Contains('.') ? UserName?.Split('.')?[1] : null;
+            return UserName.Contains('.')?UserName?.Split('.')?[1] : null;
         }
 
+        public bool NicknameMatches(string matchString)
+        {
+            return Nickname?.ToLower() == matchString;
+        }
 
+        public bool NicknameStartsWith(string searchString)
+        {
+            return Nickname?.ToLower()?.StartsWith(searchString) ?? false;
+        }
+
+        public bool NicknameContains(string searchString)
+        {
+            return Nickname?.ToLower()?.Contains(searchString) ?? false;
+        }
     }
 
```

Comment:
Using prettier here would help with the spacing at the end of the code.

Reply:
That was the original code before my modification.